resources/js/components/fieldtypes/QueryBuilder.vue

```diff
@@ -0,0 +1,417 @@
+                    { value: 'NEXT_X_DAYS', label: 'In the next days' },
+                    { value: 'THIS_WEEK', label: 'This week' },
+                    { value: 'THIS_MONTH', label: 'This month' },
+                    { value: 'THIS_YEAR', label: 'This year' },
+                    { value: 'IS_NULL', label: 'No date set' },
+                    { value: 'IS_NOT_NULL', label: 'Has date' }
+                ]
+            })
+        },
+        defaultLimit: {
+            type: Number,
+            default: 100
+        },
+        showLimit: {
+            type: Boolean,
+            default: true
+        }
+    },
+
+    data() {
+        return {
+            groups: [],
+            limit: this.defaultLimit,
+            globalConjunction: 'AND',
+            logicalOperators: ['AND', 'OR']
+        }
+    },
+
+    methods: {
+        initializeGroups() {
+            if (this.value?.groups?.length) {
+                return this.value.groups;
+            }
+            return [{
+                conjunction: 'AND',
+                conditions: []
+            }];
+        },
+
+        addGroup() {
+            this.groups.push({
+                conjunction: 'AND',
+                conditions: []
+            });
+            this.updateValue();
+        },
+
+        removeGroup(groupIndex) {
+            this.groups.splice(groupIndex, 1);
+            this.updateValue();
+        },
+
+        addConditionToGroup(groupIndex) {
+            const defaultOperator = this.getOperatorsForType(this.fields[0]?.value)[0]?.value || '=';
+            const defaultValue = this.needsBetweenInput(defaultOperator) ? { min: '', max: '' } : '';
+            
+            this.groups[groupIndex].conditions.push({
+                attribute: this.fields[0]?.value || '',
+                operator: defaultOperator,
+                value: defaultValue
+            });
+        },
+
+        removeCondition(groupIndex, conditionIndex) {
+            this.groups[groupIndex].conditions.splice(conditionIndex, 1);
+            this.updateValue();
+        },
+
+        getField(fieldValue) {
+            return this.fields.find(f => f.value === fieldValue);
+        },
+
+        getFieldOperators(fieldType) {
+            return this.operators[fieldType] || this.operators.text;
+        },
+
+        getFieldOptions(fieldValue) {
+            const field = this.fields.find(f => f.value === fieldValue);
+            return field?.options || [];
+        },
+
+        isMultiSelectVisible(condition) {
+            const field = this.fields.find(f => f.value === condition.attribute);
+            return field?.type === 'select' && ['IN', 'NOT IN'].includes(condition.operator);
+        },
+
+        isSingleSelectVisible(condition) {
+            const field = this.fields.find(f => f.value === condition.attribute);
+            return field?.type === 'select' && !['IN', 'NOT IN'].includes(condition.operator);
```

Comment:
I'm not really a fan of single character variables but that's personal, it doesn't really describe what it is even though it's clear in this case